pkg/block/indexheader/lazy_binary_reader.go

```diff
@@ -268,6 +276,20 @@ func (r *LazyBinaryReader) unload() error {
 	return nil
 }
 
-func (r *LazyBinaryReader) lastUsedAt() int64 {
-	return r.usedAt.Load()
+// isLoaded returns true if the underlying index-header reader is currently loaded.
+func (r *LazyBinaryReader) isLoaded() bool {
+	r.readerMx.RLock()
```

Comment:
I think this function is nice if it would be used more times than once, otherwise it might be just confusing. (It brings question why one method is mutex, another not).

Maybe inlining it would be better?

Not a blocker.

Reply:
Good point. Inlined.